multimodal/vl2l/src/mlperf_inference_multimodal_vl2l/evaluation.py

```diff
@@ -107,22 +108,40 @@ def calculate_hierarchical_f1(data: list[tuple[str, str]]) -> float:
     return 0.0 if hp + hr == 0 else 2 * (hp * hr) / (hp + hr)
 
 
-def calculate_exact_match(generated_text: str, original_text: str) -> float:
-    """Calculates binary Exact Match (EM) score.
-
-    We clean the text (lowercase, strip whitespace) for a fairer comparison.
+def calculate_brand_f1_score(data: list[tuple[str, str]]) -> float:
+    """Calculate the F1 score of brand field.
 
     Args:
-        generated_text: Output from the VLM.
-        original_text: Ground truth information from the dataset.
+        data: A list of tuples, where each tuple is
+            (predicted_path_str, true_path_str).
 
     Returns:
-        1 if the values match or 0 otherwise
+        F1 score
     """
-    gen = generated_text.strip().lower()
-    orig = original_text.strip().lower()
+    valid_threshold = 90
+    matches = []
+    for pred, src in data:
+        norm_truth = src.strip().lower()
+        norm_pred = pred.strip().lower()
 
-    return 1.0 if gen == orig else 0.0
+        # Exact Match
+        if norm_truth == norm_pred:
+            matches.append(1)
+            continue
```

Comment:
I'm just wondering, if it's an exact match, wouldn't the score from `fuzz.ratio` be also bigger than `valid_threshold`? Therefore, maybe there's no need to treat the exact match as a special case (that needed to be handled differently)?